include/tvm/runtime/disco/session.h

```diff
@@ -236,7 +248,7 @@ class Session : public ObjectRef {
  public:
   /*! \brief Create a session backed by a thread pool of workers */
   static Session ThreadedSession(int num_workers);
-  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(Session, ObjectRef, SessionObj);
+  TVM_DEFINE_MUTABLE_OBJECT_REF_METHODS(Session, ObjectRef, SessionObj);
```

Comment:
Why this change?

Reply:
In MLC we want to make Session a field. To initialize it, we are using ThreadSession(worker_num) instead of its constructor to create a session, so it has to be nullable to compile. Another reason is that in definition of DRef, we are already having  
``` /*! \brief Back-pointer to the host controler session */
  ObjectRef session{nullptr};
```
So there's no reason session can't be nullable.

Reply:
To be clear, it's not-nullable since it's properly created. If you wanted to cover the case where Session could be a field of another TVM Object, the common paradigm we are using across TVM codebase is to mark it as "initially null":

```C++
Session sess{nullptr};
```

Note that this pattern is widely used in MetaSchedule already.

There are multiple folds of reasons for using not-nullable, as TVM object system will enforce more checking, and also developers of other modules could feel very safe to use an object without worrying it's null.

Reply:
sounds good